gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ConditionMapper.java

```diff
@@ -303,11 +304,26 @@ private List<RCMRMT030101UK04EhrComposition> getCompositionsContainingLinkSets(R
     }
 
     private Optional<RCMRMT030101UK04ObservationStatement> getObservationStatementById(RCMRMT030101UK04EhrExtract ehrExtract, String id) {
-        return ehrExtract.getComponent().get(0).getEhrFolder().getComponent().stream()
-            .flatMap(e -> e.getEhrComposition().getComponent().stream())
-            .map(RCMRMT030101UK04Component4::getObservationStatement)
-            .filter(Objects::nonNull)
-            .filter(observationStatement -> id.equals(observationStatement.getId().getRoot()))
-            .findFirst();
+        var componentsByHasCompoundStatement = ehrExtract.getComponent()
+                .get(0)
+                .getEhrFolder()
+                .getComponent()
+                .stream()
+                .flatMap(e -> e.getEhrComposition().getComponent().stream())
+                .collect(partitioningBy(RCMRMT030101UK04Component4::hasCompoundStatement));
+
+
+        var obsList = Stream.concat(componentsByHasCompoundStatement.get(false)
```

Comment:
Should this not be observationList by clean coding standards? (No abbreviations)